### PR TITLE
Fix large_metadata_bad_client_test to avoid C99 string literal length limit.

### LINKED ISSUE #6551: Too much metadata tests break -Woverlength-strings
test/core/bad_client/tests/large_metadata.c has an oversized string literal defined in the macro PFX_TOO_MUCH_METADATA_FROM_CLIENT_STR.

This triggers a warning in our internal build:
test/core/bad_client/tests/large_metadata.c:481:28: error: string literal of length 13733 exceeds maximum length 4095 that ISO C99 compilers are required to support [-Werror,-Woverlength-strings]

### PR DESCRIPTION
**Pull request by markdroth**
Fixes #6551.

### CODE REVIEW

File: test/core/bad_client/tests/large_metadata.c [hunk 1]
```diff
@@ -477,8 +217,19 @@ int main(int argc, char **argv) {
   grpc_test_init(argc, argv);
 
   // Test sending more metadata than the server will accept.
+  char client_payload[PFX_TOO_MUCH_METADATA_FROM_CLIENT_PAYLOAD_SIZE] =
+      PFX_TOO_MUCH_METADATA_FROM_CLIENT_PREFIX_STR;
+  size_t offset = sizeof(PFX_TOO_MUCH_METADATA_FROM_CLIENT_PREFIX_STR);
+  for (int i = 0; i < NUM_HEADERS; ++i) {
+    snprintf(client_payload + offset,
```
Comment (ctiller) [inline test/core/bad_client/tests/large_metadata.c]: Prefer gpr_asprintf (it's at least supported everywhere)

Comment (markdroth) [inline test/core/bad_client/tests/large_metadata.c]: Done.

**Comment by dklempner** [comment]
LGTM

**Comment by dklempner** [comment]
"Basic tests" failures are obviously unrelated, so I'll just merge this.

**Pull request merged**